.github/workflows/codecov.yml

```
@@ -0,0 +1,51 @@
+name: Codecov
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    types: [opened, reopened, synchronize]
+
+jobs:
+  codecov-grcov:
+    name: Generate code coverage
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: beta # temporarily use beta till 'instrument-coverage' stabilizes in 1.60
+          profile: default
+          override: true
+          components: llvm-tools-preview
```

Lets drop the requirement on the beta toolchain and instead use whats in the `rust-toolchain` file (which can be updated to 1.60 if it isn't already).
```suggestion
      - uses: actions-rs/toolchain@v1
        with:
          components: llvm-tools-preview
```

---

This is due to coverage feature being stable in 1.60